### PR TITLE
AN-268 GCP Batch subnetwork wildcard support

### PR DESCRIPTION
**Pull request by jgainerdewar**
### Description

Adds wildcard support for regions in subnet config. This is required for VPCs with custom subnetwork creation.

Also fixes an issue where, when the user supplied multiple `zones` in WDL runtime attributes, only the first was used.

#### `CHANGELOG.md`
 - [x] I updated `CHANGELOG.md` in this PR
 - [ ] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [x] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

File: supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/api/GcpBatchRequestFactoryImpl.scala [hunk 1]
```diff
@@ -56,8 +56,10 @@ class GcpBatchRequestFactoryImpl()(implicit gcsTransferConfiguration: GcsTransfe
       .setNoExternalIpAddress(data.gcpBatchParameters.runtimeAttributes.noAddress)
       .setNetwork(vpcAndSubnetworkProjectLabelValues.networkName(data.createParameters.projectId))
 
+    val region = zonesToRegion(data.createParameters.runtimeAttributes.zones).getOrElse(data.gcpBatchParameters.region)
```
Comment (jgainerdewar) [inline supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/api/GcpBatchRequestFactoryImpl.scala:59]: Confirmed in manual testing that `gcpBatchParameters.region` will be the workspace region, not the base GCP Batch region.

**Comment by LizBaldo** [comment]
Also, not quite sure what is going on with the integration tests, but it is interesting that they are already running with MySQL 5.7 🤯

**Comment by mcovarr** [comment]
Can there be / should there be a Centaur test as this VPC configuration is actually what's used in Terra?

**Comment by aednichols** [comment]
@LizBaldo we've typically tested against the most recent MySQL and the oldest supported one – because operators outside Terra might be ahead of us.

It looks like maybe "oldest supported" has left 5.6 behind and advanced to 5.7. Something to look at in [that one ticket](https://broadworkbench.atlassian.net/browse/AN-270).

File: supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/api/GcpBatchRequestFactoryImpl.scala [hunk 2]
```diff
@@ -252,7 +254,7 @@ class GcpBatchRequestFactoryImpl()(implicit gcsTransferConfiguration: GcsTransfe
     )
     val instancePolicy =
       createInstancePolicy(cpuPlatform = cpuPlatform, spotModel, accelerators, allDisks, machineType = machineType)
-    val locationPolicy = LocationPolicy.newBuilder.addAllowedLocations(zones).build
+    val locationPolicy = LocationPolicy.newBuilder.addAllAllowedLocations(zones.asJava).build
```
[the remark this answers is not in the document]
Comment (jgainerdewar) [reply supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/api/GcpBatchRequestFactoryImpl.scala:257]: `LocationPolicy` comes from the google SDK that we're working with, which is in Java - so it wants us to pass in a Java `Iterable<String>`.

**Pull request merged**